lib/chef/knife/azurerm_base.rb

```diff
@@ -146,18 +174,18 @@ def check_token_validity(token_details)
 
       def validate_azure_login
         err_string = "Please run XPLAT's 'azure login' command OR specify azure_tenant_id, azure_subscription_id, azure_client_id, azure_client_secret in your knife.rb"
-        if Chef::Platform.windows?
+        if Chef::Platform.windows? && (is_old_xplat? || is_WCM_env_var_set?)
           # cmdkey command is used for accessing windows credential manager
           xplat_creds_cmd = Mixlib::ShellOut.new("cmdkey /list | findstr AzureXplatCli")
           result = xplat_creds_cmd.run_command
           if result.stdout.nil? || result.stdout.empty?
             raise err_string
           end
-        else
-          home_dir = File.expand_path('~')
-          if !File.exists?(home_dir + "/.azure/accessTokens.json") || File.size?(home_dir + '/.azure/accessTokens.json') <= 2
-            raise err_string
-          end
+          return
```

Comment:
@btm Here, the `else` part has now become common for both Windows and Linux platforms (earlier `if` part was only for Windows and `else` part was only for Linux platform). In Windows case the `else` part will be used if `AZURE_USE_SECURE_TOKEN_STORAGE` environment variable is not set. In the other case if this environment variable is set then the `if` part will be used (like earlier) and `else` part will be skipped with `return` statement. In Linux case, only `else` part will be used.